### PR TITLE
Initial draft of meta data element from SSP Traceability

### PR DESCRIPTION
**Pull request by pmai**
This is the basic idea of porting the meta data container from SSP Traceability to SSD. Needs more refinement, and addition to other places, but should complement adding classifications for inline info vs. refering to external or different meta-data formats.

### CODE REVIEW

**Comment by JochenKoehler** [comment]
Plan to discuss on April, 22. Dag and Klaus will review it (also Peter)

**Review by DagBruck — APPROVED** [review]

**Comment by JochenKoehler** [comment]
web meeting 2024-04-26
No reviews done so far - still pending

**Comment by peter-lobner** [comment]
Personally I would vote to remove the signature stuff, as we don't have this for anything else in the SSP Standard currently.

**Comment by JochenKoehler** [comment]
Web meeting 2024-05-08
Discussion about adding signature.
Decision is pending where to add also the possibility to "attach" Meta data. And during this process it could be decided to add also signature for these parts.
More use of "complex type"? - Pierre will have some thoughts about it

**Comment by pmai** [comment]
Added meta-data to the other places that reference resources; this raises a couple of questions, though:

- The complex type for in-line content can't be reused, as most places where it would be used have additional content/checks, especially in the 1.1 schema.
- As a couple of places directly inline the content (e.g. ParameterMapping, vs. ParameterBinding that has a separate element for the in-line content ParameterValues), this leads to the awkward situation that the MetaData element sits in the same place as the potential inline content; this really can't be expressed in 1.0 schema, and might still be a bit jarring in the 1.1 schema. 
- Relatedly, we might want to allow mixed-content (i.e. arbitrary text without any wrapping XML element) in some of those places, but that again does not work nicely if there is no wrapping element.

Changing any of this however would likely break back-ward compatibility and/or make the schema and handling more complex. So this is likely something that should be discussed at the F2F.

File: schema/SystemStructureCommon.xsd [hunk 1]
```diff
@@ -255,6 +255,191 @@
...
+                                <xs:enumeration value="general"/>
+                                <xs:enumeration value="quality"/>
+                            </xs:restriction>
+                        </xs:simpleType>
+                    </xs:attribute>
+                    <xs:attribute name="type" type="xs:string" use="required">
+                        <xs:annotation>
+                            <xs:documentation xml:lang="en">
+                                This mandatory attribute specifies the MIME type of the resource meta data, which
+                                does not have a default value.  If no specific MIME type can be indicated, then
```
Comment (pmai) [inline schema/SystemStructureCommon.xsd:305]: F2F 2024-06-12: Should maybe default to SRMD MIME Type? Better not, as that will likely lead to many non-SRMD files being attached but with the default MIME Type, which leads to problems.

**Comment by pmai** [comment]
F2F 2024-06-12: Approach is sound, @pmai to add documentation changes based on the current proposal. Signature stuff stays in for now, as this would allow re-use of the definitions in the SSP Traceability Layer as-is. Decision might be revised prior to release if needed.

**Review by pmai — COMMENTED** [review]
CCB 2024-09-13: Signature element should also be added at the GMetaData level to allow signatures for the resources themselves and not only for their meta data. @pmai to make changes and PR to be rediscussed in next meeting for final merge.

File: docs/4___common_content.adoc [hunk 2]
```diff
@@ -460,3 +460,112 @@ This attribute gives the size of this dimension of the array as a fixed, unchang
...
+|Attribute |Description
+|kind a|
+This attribute indicates the kind of model element meta data that is referenced, i.e. what role it plays in relation to the model element being described.
+The kind can be `general` or `quality`, indicating general meta data or quality related meta data, respectively.
+|type a|
+This mandatory attribute specifies the MIME type of the meta data, which does not have a default value.
+If no specific MIME type can be indicated, then the type `application/octet-stream` is to be used.
+|source a|
+Optional attribute indicating the source of the meta data as a URI (cf. RFC 3986).
+For purposes of the resolution of relative URIs the base URI is the URI of the SSD, if the sourceBase attribute is not specified or is specified as `SSD`, and the URI of the containing model element (e.g. component) if the base attribute is specified as `component`.
```
Comment (klausschuch) [inline docs/4___common_content.adoc:511]:
```suggestion
The sourceBase attribute is used for the resolution of relative URIs.
If the sourceBase attribute is not specified or is specified as `SSD`, the base URI is the URI of the SSD.
If the sourceBase attribute is specified as `component`, the base URI is the URI of the containing model element (e.g. component).
```

Comment (pmai) [reply docs/4___common_content.adoc:511]: 2024-09-27: Should be clarified by removing most of the duplicated content from sourceBase. This is to be handled in a separate PR which addresses the same language occuring in all instances of source/sourceBase.

**Comment by pmai** [comment]
CCB 2024-09-27: GMetaData to be added at top-level to all other file formats, then PR can be merged by @pmai.

**Comment by peter-lobner** [comment]
Hi @pmai,

it would be great if GMetaData would be also availale at parameter level.

**Pull request merged**